tests/Functional/Level1/UnspecifiedDigitTest.php

```diff
@@ -17,18 +18,25 @@ class UnspecifiedDigitTest extends TestCase
 {
     use FactoryTrait;
 
-    public function testWithUnspecifiedOneOrTwoDigitYear()
+    /**
+     * @dataProvider unspecifiedYear
+     * @param string $input
+     * @param int $expectedYear
+     * @param int $expectedMin
+     * @param int $expectedMax
+     * @throws \EDTF\Exceptions\ExtDateException
+     */
+    public function testWithUnspecifiedYear(string $input, int $expectedYear, int $expectedMin, int $expectedMax)
     {
-        $d = $this->createExtDate('201X');
+        $d = $this->createExtDate($input);
         $this->assertTrue($d->unspecified());
         $this->assertTrue($d->unspecified('year'));
         $this->assertFalse($d->unspecified('month'));
-        $this->assertSame(2010, $d->getYear());
+        $this->assertFalse($d->unspecified('day'));
+        $this->assertSame($expectedYear, $d->getYear());
 
-        $d = $this->createExtDate('20XX');
-        $this->assertTrue($d->unspecified());
-        $this->assertTrue($d->unspecified('year'));
-        $this->assertSame(2000, $d->getYear());
+        $this->assertEquals($expectedMin, $d->getMin());
+        $this->assertEquals($expectedMax, $d->getMax());
```

Comment:
When dealing with scalars using assertsSame (===) is generally better than assertEquals (==)